pkg/agent/goclient_opensearch.go

```diff
@@ -0,0 +1,169 @@
+package agent
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/pprof/profile"
+	"github.com/joho/godotenv"
+	opensearch "github.com/opensearch-project/opensearch-go"
+	opensearchapi "github.com/opensearch-project/opensearch-go/opensearchapi"
+)
+
+var IsInsecure = false
```

Comment:
is this always false? if not where is this being set?

Reply:
Set in main.go by reading the flag passed on the command line